clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -65,7 +80,27 @@ class ScalarExprEmitter : public StmtVisitor<ScalarExprEmitter, mlir::Value> {
         cgf.getLoc(e->getExprLoc()), type,
         builder.getCIRBoolAttr(e->getValue()));
   }
+
+  mlir::Value VisitCastExpr(CastExpr *E);
+
+  /// Emit a conversion from the specified type to the specified destination
+  /// type, both of which are CIR scalar types.
+  /// TODO: do we need ScalarConversionOpts here? Should be done in another
+  /// pass.
+  mlir::Value emitScalarConversion(mlir::Value src, QualType srcType,
+                                   QualType dstType, SourceLocation loc) {
+    // No sort of type conversion is implemented yet, but the path for implicit
+    // paths goes through here even if the type isn't being changed.
+    srcType = srcType.getCanonicalType();
+    dstType = dstType.getCanonicalType();
+    if (srcType == dstType)
+      return src;
+
+    cgf.getCIRGenModule().errorNYI(loc,
+                                   "emitScalarConversion for unequal types");
```

Comment:
I missed this last time I went through this, but since errorNYI doesn't terminate, this falls off the end of the function!  Please commit something that puts a return after this (feel free to review-after-commit).